### PR TITLE
Make DBUS session check more resilient

### PR DESCRIPTION
**Pull request by eloquence**
We should always ensure that the DBUS session is set to the logged in user, who should not be root.

Towards #668 

## Status

Ready for review

## Test plan

1. Configure your environment to `staging`
2. Check out this branch in `dom0` via `make clone`
3. Build a staging env via `make staging`
4. - [ ] Observe that `update-xfce-settings` runs successfully
5. Attempt to run the `update-xfce-settings` script as root
6. - [ ] Observe that you can't

### CODE REVIEW

**Review by conorsch — APPROVED** [review]
Works as advertised. Still can't explain how uid 0 was running those tasks, but the error report was clear that it was happening.

Spent some time reading up on how dbus works, which has changed a bit in versions of systemd, so we may want to revisit this code during the Qubes 4.1 transition (when systemd should go from v231 to v245). I'll post a patch that I tested locally, but am not appending to the PR as part of the merge, since the changes presented work well as they are.

**Comment by eloquence** [comment]
> but the error report was clear that it was happening.

Just noting that, if you look at @emkll's reported output,  you'll notice that $USER was correctly set to `m` (his Qubes username). My hypothesis is that `runas` was working just fine, but for some reason (use of `sudo` or something else) there was also a DBUS session for root, which the script as previously written did not override.

**Comment by conorsch** [comment]
Tried this locally: 

```patch
diff --git a/dom0/update-xfce-settings b/dom0/update-xfce-settings
index f1adecf..4d64730 100755
--- a/dom0/update-xfce-settings
+++ b/dom0/update-xfce-settings
@@ -23,12 +23,15 @@ ICONSIZE=64
 
 if ! [ -x "$(command -v xfconf-query)" ]; then
   echo "Error: xfconf-query is not installed." >&2
-  exit 1
+  exit 2
 fi
 
-# This script requires a valid DBUS session to work. When run non-interactively,
-# we assume that a sesssion is running for the current user.
-export DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/$(id -u $USER)/bus"
+# This script requires a valid DBUS session to work. When run interactively,
+# DBUS_SESSION_BUS_ADDRESS should already be populated. Otherwise, create
+# a session so that the xfconf-query commands succeed.
+if [[ -z "${DBUS_SESSION_BUS_ADDRESS:-}" ]]; then
+    eval "$(dbus-launch)"
+fi
``` 

and it handled the edge case of DBUS_SESSION_BUS_ADDRESS not being defined. Which notably it isn't during a `make staging` run: if you `test -n $DBUS_SESSION_BUS_ADDRESS` in that script, it will fail. Whether we hardcode the path or politely request a session doesn't seem to make much of a difference. Once downside of politely requesting a fresh session via dbus-launch is that the session appears to hang around forever. In the case of the Workstation, we expect shutdowns almost daily, but I'm still not comfortable forking the dbus process on every update without identifying a cleanup mechanism.
